### DISCUSSION TITLE
Issue with Building an Executable from Python Script Using PyInstaller

### DISCUSSION THREAD

**Discussion by wy8933** [LyubomirT/wincuts#9] [Q&A] [answered]
Hello, everyone. I am new to open source and don't have much experience when building exe. I'm encountering a problem building an exe from my Python script using PyInstaller. The script runs perfectly within my IDE (PyCharm), but the program closes instantly when I create an executable and try to run it. This causes me to be unable to test the uninstaller I wrote for issue#2.

The command I used to build the executable: 
pyinstaller --onefile wincuts.py
pyinstaller  wincuts.py
pyinstaller --onefile uninstaller.py
pyinstaller uninstaller.py

What I have tried:
I've confirmed that all dependencies are correctly installed in my environment.
I've tested the script in the IDE to ensure it runs without errors.
I've attempted to build the executable with and without the --onefile option.

Does anyone have idea on what might be going wrong? The script runs without issues in the IDE, but not as a built executable. Any advice, insights, or suggestions would be greatly appreciated.

Thank you in advance for your time and help!

**Comment by LyubomirT** [comment, marked as answer]
@wy8933 Hey there. The build command I used for the wincuts.py file is:

```shell
pyinstaller --noconfirm --onedir --windowed  "wincuts.py"
```

I use `--onedir` here as PyInstaller is quite limited in terms of functionality of the built `.exe` if in `--onefile`. Then I also place the `logo.png` in the same directory as `wincuts.exe`. Afterwards, I believe it should work. As for your uninstaller, I sadly can't see the code of it for now, and I can't decide appropriately. Although, if you want to take some local files, that will **not** work in `--onefile` (that's why we have `integration.exe` with `--onedir`).

**Reply by wy8933** [reply]
Thank you. The application can now be opened.

**Reply by LyubomirT** [reply]
Happy to help! :)
